### PR TITLE
sdl2_mixer: add upstream patch for use-after-free bug

### PR DESCRIPTION
**Pull request by Kreeblah**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/sdl2_mixer.rb [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
-  bottle do
-    sha256 cellar: :any,                 arm64_monterey: "375fbfb9b64fdfc631a7446aa1e21c74f3fc1b5a2e732cbe9db711078fe9c805"
-    sha256 cellar: :any,                 arm64_big_sur:  "f3c822ba6f80353d0fee40ea106025d4519e1dbb141d097fef092393e95fb0b7"
-    sha256 cellar: :any,                 monterey:       "4784c901a5fe2da26d58b06d1c888cb861874cb16d1211705066dcd7a61027ab"
-    sha256 cellar: :any,                 big_sur:        "1529a00916c4066d8adc0987b627e2bc7cf66aca063562cb3af64f8fa5f231f7"
-    sha256 cellar: :any,                 catalina:       "9779416544a0d71a8206b45895a3060baca2bf0877441017aaa6b1d6136654a2"
-    sha256 cellar: :any,                 mojave:         "9c13dd597aca2e0d5f53f2a7b4a1ea4e5a724c08796ba0eaf71a54f9cc714fbc"
-    sha256 cellar: :any_skip_relocation, x86_64_linux:   "412de036ef39cab5f7e1907711c2273a3d671e995ddeda320cad9cfedaa72a28"
-  end
-
```
Comment (carlocab) [inline Formula/sdl2_mixer.rb:23]: Please don't touch the bottle block.

**Review by carlocab — COMMENTED** [review]
Thanks, @Kreeblah. Can you reinstate the `bottle` block? When you do so, please amend your existing commit (i.e. don't add any new ones), and change the commit message to
```
sdl2_mixer: add upstream patch for use-after-free bug
```

We put the name of the formula first in commit-message headings. Refer to the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit) for more details.

File: Formula/sdl2_mixer.rb (hunk 1 above)
Comment (Kreeblah) [reply Formula/sdl2_mixer.rb]: Ah.  I wasn't sure how that worked with moving things around in the formula.  I put that back in.

**Comment by Kreeblah** [comment]
Sure.  I just made those changes.

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request closed without merging**